### ISSUE TITLE
Add Ed25519-SHA256 signing support?

### ISSUE THREAD

**Issue by Deckard99** [Pro/dkim-exchange#264]
Hi,

anyone able to integrate the new EdDSA  signing algorithm into the agent?
https://tools.ietf.org/html/rfc8463

regards
Norbert

**Comment by jstedfast** [comment]
I've just implemented rfc8463 as part of [MimeKit](https://github.com/jstedfast/MimeKit) here: https://github.com/jstedfast/MimeKit/commit/9c7ef4426650081ee1c3afeef2b49a1896383d53

This will eventually be released as part of MimeKit 2.2.0.

A lot of my current work in MimeKit revolves around implementing ARC support which, naturally, involves refactoring of DKIM support and with that I've noticed some of the DKIM updates...

Anyway, that said... I'll be moving into a new house soon which will kill my free time to work on this for a while, so I'm not sure when 2.2.0 can be expected.

**Comment by Deckard99** [comment]
Thanks for the info. Well I can wait. Good luck with the move into your new house.

**Comment by sarthurdev** [comment]
Hi @Deckard99 

I have created a branch for this (https://github.com/Pro/dkim-exchange/tree/Ed25519-SHA256). It is untested as of yet.

If you have a test Exchange environment you can use, any help with testing would be greatly appreciated.

**Comment by Deckard99** [comment]
Hi,
I installed it on an Edge 2016 (CU13) server. After initial configuration (running 3.2 before) the transport service will not start.
Event 1016 The worker process crashes continuously on startup: C:\Program Files\Microsoft\Exchange Server\V15\Bin\edgetransport.exe. The service will be stopped.

**Comment by Deckard99** [comment]
Btw. Wouldn't it be good to have the ability to sign twice (RSA and Ed25519) for older systems which probably won't handle Ed25519 signed mails?

**Comment by sarthurdev** [comment]
Probably, but I don't have too much time to sink into this at the moment. Will setup a test environment for this branch next week and see if I can at least fix the transport crash.

**Comment by Deckard99** [comment]
No problem. Thanks for your contribution here.

**Comment by rhymeswithmogul** [comment]
> Btw. Wouldn't it be good to have the ability to sign twice (RSA and Ed25519) for older systems which probably won't handle Ed25519 signed mails?

Correct.  Because you only need one valid signature for `dkim=pass`, emails ought to be doubly signed for the foreseeable future, once with Ed25519 and once with RSA.  Newer verifiers will check the Ed25519 signature first (and if it doesn't verify, fall back to RSA); however, older verifiers will ignore the Ed25519 signature and only check the RSA one.

Ideally, dkim-exchange would let the admin generate RSA and/or Ed25519 keys, then offer the option to sign emails with RSA and/or Ed25519.  Dual-signing should be the default option for new installs, in my opinion.

**Comment by Deckard99** [comment]
Hi,
just want to ask if there is any progress on this topic? If any testers are needed just let me know.
